src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderSettingsDao.java

```diff
@@ -65,18 +81,47 @@ public SnapshotBuilderSettings upsertSnapshotBuilderSettingsByDataset(
             + " VALUES (:dataset_id, cast(:settings as jsonb))"
             + " ON CONFLICT ON CONSTRAINT snapshot_builder_settings_dataset_id_key"
             + " DO UPDATE SET settings = cast(:settings as jsonb)",
-        Map.of(datasetIdField, datasetId, "settings", jsonValue));
+        Map.of(DATASET_ID_FIELD, datasetId, SETTINGS_FIELD, jsonValue));
     return getSnapshotBuilderSettingsByDatasetId(datasetId);
   }
 
   @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
-  public void delete(UUID datasetId) {
+  public SnapshotBuilderSettings upsertSnapshotBuilderSettingsBySnapshotId(
```

Comment:
I think since we only need either a dataset ID or a snapshot ID it makes sense to keep them separate. But I want to better understand this suggestion, why would we want one method that takes both/what is the benefit?

Reply:
I agree, I like them split because we can more easily remove them later.